tests/unit/s2n_client_hello_recv_test.c

```diff
@@ -405,7 +403,11 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_client_hello_send(client_conn));
             EXPECT_SUCCESS(s2n_stuffer_copy(&client_conn->handshake.io, &server_conn->handshake.io,
                     s2n_stuffer_data_available(&client_conn->handshake.io)));
-            EXPECT_SUCCESS(s2n_client_hello_recv(server_conn));
+            if (s2n_is_tls_12_self_downgrade_required(client_conn)) {
+                EXPECT_FAILURE_WITH_ERRNO(s2n_client_hello_recv(server_conn), S2N_ERR_PROTOCOL_VERSION_UNSUPPORTED);
+            } else {
+                EXPECT_SUCCESS(s2n_client_hello_recv(server_conn));
+            }
```

Comment:
This test isn't interested in fallback behavior. TLS1.2 + QUIC always fails, but we're not testing that here. Probably worth restricting it to TLS1.3 only. Same for the next one.

Reply:
Fixed